packages/astro/src/content/runtime.ts

```diff
@@ -433,7 +440,14 @@ function updateImageReferencesInData<T extends Record<string, unknown>>(
 	});
 }
 
-async function renderEntry(entry?: DataEntry) {
+export async function renderEntry(
+	entry?: DataEntry | { render: () => Promise<{ Content: AstroComponentFactory }> }
```

Comment:
It feels weird to have a public function called `renderEntry` where the `entry` is optional

Reply:
Yeah same, if we are making this public we can refactor the API just a tad.